### PR TITLE
Bump golang builder to v1.17

### PR DESCRIPTION
**Pull request by btaani**
### Description
This PR upgrades Golang to v1.17 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-2330

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### This is a generated file from Dockerfile.in ###
 #@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.16)
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.8 AS builder
```
Comment (xperimental) [inline Dockerfile:3]: I think you need to update the "follow_tag" line as well. For the loki build we switched to "4.10" in the base image. The image you are now referring to does not seem to exist, so maybe 4.10 is the first version with Go 1.17.

```suggestion
#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.17)
FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10 AS builder
```

The suggestion display seems wonky (one line too much), don't know if that UI feature will work here...

File: Dockerfile.dev [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ### This is a generated file from Dockerfile.in ###
 #@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.16)
-FROM registry.ci.openshift.org/openshift/release:golang-1.16 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
```
Comment (xperimental) [inline Dockerfile.dev:3]: The `follow_tag` line needs an update.

File: origin-meta.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from:
 - source: registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder\:v(?:[\.0-9\-]*).*
-  target: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
+  target: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.8 AS builder
```
Comment (xperimental) [inline origin-meta.yaml:3]: Need to update to `4.10` as well.

**Review by xperimental — CHANGES_REQUESTED** [review]
See inline comments.

Also, I found one file which has a `+build` line which needs to be updated for Go 1.17. This can be done automatically using `go fmt`.

File: Dockerfile (hunk 1 above)
Comment (btaani) [reply Dockerfile:3]: `go fmt` does not seem to do anything for me, so I manually fixed the `follow_tag`

File: Dockerfile.in [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.16)
+#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.17)
 FROM registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:v1.16.6-202107291435.el8.git.83c4106 AS builder
```
[the remark this answers is not in the document]
Comment (btaani) [reply Dockerfile.in:2]: Thanks @xperimental 
But how is this extended version number generated?

**Review by xperimental — APPROVED** [review]

**Review by periklis — COMMENTED** [review]
Good work :rocket: Mostly what @xperimental commented above.

**Pull request merged**